From f68c3fd2861167df17b95985014867dd4eaa445b Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 4 Mar 2025 18:18:07 +0100
Subject: [PATCH] arm64: dts: st: select AM clock on LPTIMER-3 on
 stm32mp257f-ev1 board

The CK_LPTIMER3 is stopped in LPLV-Stop1 mode, so to support CpuIdle PSCI
OSI with LP-TIMER3 must use the autonomous mode of the clock by selecting
the RCC CK_LPTIM3_AM clock.

Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I55246e388261daa3083c0bec1a9b278b4f4b22ca
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/441793
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -502,6 +502,7 @@
 
 /* use LPTIMER with tick broadcast for suspend mode */
 &lptimer3 {
+	clocks = <&rcc CK_LPTIM3_AM>;
 	status = "okay";
 	timer {
 		status = "okay";
